lib/bootloader_setup.pm

```diff
@@ -1040,37 +1040,95 @@ sub change_grub_config {
 
 =head2 add_grub_cmdline_settings
 
-    add_grub_cmdline_settings($add [, $update_grub ]);
+    add_grub_cmdline_settings($add [, $update_grub , $search]);
 
 Add C<$add> into /etc/default/grub, using sed.
 C<$update_grub> if set, regenerate /boot/grub2/grub.cfg with grub2-mkconfig and upload configuration.
+C<$search> if set, bypass default grub cmdline variable.
 =cut
 sub add_grub_cmdline_settings {
+    my (%args) = compat_args(
+        {
+            add         => '',
+            update_grub => 0,
+            search      => get_cmdline_var();
+        }, ['add', 'update_grub', 'search'], @_);
```

Comment:
Good point to use `compat_args` :).

Reply:
but compat_args aren't exported by testapi.


It broke JeOS completely :)
https://openqa.suse.de/tests/3451320#step/grub2_gfxmode/10